### DISCUSSION TITLE
Bootstrap 5.2 - Any JS dependency changes?

### DISCUSSION THREAD

**Discussion by chrisreddington** [twbs/bootstrap#36881] [General]
Hi Folks,

I wondered if anyone had any issues upgrading from Bootstrap 5.2 to 5.2? I'm aware that there were some changes on the sass side (so have imported the new maps partial), but seem to also be having issues with JavaScript.

I can't see anything called out in the [Bootstrap 5.2 migration guide](https://getbootstrap.com/docs/5.2/migration/) around JavaScript. I had [a site](https://www.cloudwithchris.com) with JavaScript that was working in 5.1, and now no longer working in 5.2. I am now seeing ``Uncaught TypeError: Class extends value undefined is not a constructor or null`` in my console, which hints to me that there's a dependency issue.

The theme itself for my site is [hosted in my hugo creator repository](https://github.com/CloudWithChris/hugo-creator), where you'll notice the example/demo site has the exact same issue (which could imply the issue on my end, and how i'm referencing the files - but haven't quite got that stage of identifying the issue yet!)

I'm referencing the individual files from node_modules, via the js/dist folder. The order that I'm referencing is -

"dom/data.js",
"dom/event-handler.js",
"dom/manipulator.js",
"dom/selector-engine.js",
"popper.js",
"base-component.js",
"carousel.js",
"collapse.js",
"dropdown.js"

Is there anything that I may need to consider going into 5.2 about the ordering of references for the individual components, or any dependency conflicts with other packages that may have been introduced?

Thanks!

**Comment by GeoSot** [comment]
You are probably referring to this change. https://github.com/twbs/bootstrap/pull/34509

We were thinking to deprecate the separated format, but as is still being used, I am trying to push a change https://github.com/twbs/bootstrap/pull/36854 which in combination with an esm build would be really handy as module.

For now if you want to move on 5.2. I suppose it would be better to use the `dist/js/bootstrap.bundle.js`
